### PR TITLE
working

### PR DESCRIPTION
**Pull request by luroche**
The first script to MANAGER.

### CODE REVIEW

File: code/MANAGER/MANAGER.py [hunk 1]
```diff
@@ -1,38 +1,140 @@
...
+            if self.__status is TERM_Option_e.FLASH_ORIG:
+                log.info(f"Flashing original program")
+                pass
 
-    def _calibStatus(self):
-        pass
+            #Configure device
+            elif self.__status is TERM_Option_e.CONF_DEV:
+                log.info(f"Configuring device")
+                self.__epc_config = term.queryEPCConf()
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: Add `ConfigureDevice` from power

File: code/MANAGER/MANAGER.py [hunk 2]
```diff
@@ -1,38 +1,140 @@
...
+            #Guided mode option
+            if guided is True:
+                option_guided = [None, TERM_Option_e.FLASH_ORIG, TERM_Option_e.CONF_DEV, TERM_Option_e.CALIB, TERM_Option_e.FLASH_CALIB, TERM_Option_e.EXIT]
+                self.__status = option_guided[option_guided.index(self.__status) + 1]
+            else:
+                self.__status = term.queryOptions()
+
+            #Flash original program
+            if self.__status is TERM_Option_e.FLASH_ORIG:
+                log.info(f"Flashing original program")
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py:108]: add `flashUC`

File: code/MANAGER/MANAGER.py [hunk 3]
```diff
@@ -1,38 +1,140 @@
...
+                        self._calibStatus(calib_mode)
+                        check_calib[calib_mode] = True
 
-    def executeMachineStatus(self):
-        pass
\ No newline at end of file
+            #Flash with calibration data
+            elif self.__status is TERM_Option_e.FLASH_CALIB:
+                log.info(f"Flashing with calibration data")
+                if self.__epc_config is None:
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: This condition should check if EPC sent info is the same as the __epc_config

File: code/MANAGER/MANAGER.py [hunk 4]
```diff
@@ -1,38 +1,140 @@
...
+                    log.error("EPC configuration not set")
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py:136]: If device not configured, configure device

File: code/MANAGER/MANAGER.py [hunk 5]
```diff
@@ -1,38 +1,140 @@
...
+                else:
+                    if False in check_calib.values():
+                        log.error("Not all calibration data are set")
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py:140]: If device not calibrate, execture calib status

File: code/MANAGER/MANAGER.py [hunk 6]
```diff
@@ -1,38 +1,140 @@
...
+                    else:
+                        pass
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: Apply flash

File: code/MANAGER/MANAGER.py [hunk 7]
```diff
@@ -1,38 +1,140 @@
...
+                if self.__epc_config is None:
+                    log.error("EPC configuration not set")
+                else:
+                    if guided is True:
+                        options_calib_mode = [PWR_Mode_e.VOLT_HS, PWR_Mode_e.VOLT_LS, PWR_Mode_e.CURR, PWR_Mode_e.TEMP_ANOD, PWR_Mode_e.TEMP_AMB, PWR_Mode_e.TEMP_BODY]
+                        for calib_mode in options_calib_mode:
+                            self._calibStatus(calib_mode)
+                            check_calib[calib_mode] = True
+                    else:
+                        calib_mode = self.__status = term.queryCalibMode()
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: Remove self.__status

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/MANAGER/MANAGER.py (hunk 1 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: The configDevice function is invoked in the init of self.__power. The function is privated.

File: code/MANAGER/MANAGER.py (hunk 2 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py:108]: Done

File: code/MANAGER/MANAGER.py (hunk 3 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Insert in TODO.

File: code/MANAGER/MANAGER.py (hunk 4 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py:136]: Done. A private function called __confDevice() has been created

File: code/MANAGER/MANAGER.py (hunk 7 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py (hunk 6 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py (hunk 5 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py:140]: Done.

File: code/MANAGER/MANAGER.py [hunk 8]
```diff
@@ -91,39 +104,48 @@ def executeMachineStatus(self) -> None:
...
                     if False in check_calib.values():
                         log.error("Not all calibration data are set")
+                        for calib in check_calib:
+                            if check_calib[calib] is False:
+                                self._calibStatus(mode = calib)
                     else:
-                        pass
+                        self.__stm.applyCalib()
+                        self.__stm.buildProject()
+                        self.__stm.flashUC()
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: add binary name: `STM32.bin`

File: code/MANAGER/MANAGER.py [hunk 9]
```diff
@@ -34,8 +34,8 @@ def __init__(self) -> None:
         self.__epc_config: STM_FLASH_Epc_Conf_c = None
         self.__status: TERM_Option_e = TERM_Option_e.CONF_DEV
         self.__power = PWR_c(config_path_file = f"") #TODO: check path
-    
-    
+        self.__stm: STM_FLASH_c = None  #TODO: check path
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: Create an default EPC_CONF and pass it to STM_Flash

File: code/MANAGER/MANAGER.py [hunk 10]
```diff
@@ -91,39 +104,48 @@ def executeMachineStatus(self) -> None:
             #Flash original program
             if self.__status is TERM_Option_e.FLASH_ORIG:
                 log.info(f"Flashing original program")
-                pass
+                if self.__epc_config is None:
+                    log.error("EPC configuration not set")
+                    self.__confDevice()
+                self.__stm.flashUC()
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: Pass org binary: `STM32_org.bin`

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/MANAGER/MANAGER.py (hunk 1 above)
Comment (mariuscrsn) [reply code/MANAGER/MANAGER.py]: Use CONFIG_WS_C.UpdatePath when device info is changed

Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py (hunk 8 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py (hunk 9 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py (hunk 10 above)
Comment (luroche) [reply code/MANAGER/MANAGER.py]: Done.

File: code/MANAGER/MANAGER.py [hunk 11]
```diff
@@ -1,38 +1,164 @@
...
+            if guided is True:
+                option_guided = [None, TERM_Option_e.FLASH_ORIG, TERM_Option_e.CONF_DEV, TERM_Option_e.CALIB, TERM_Option_e.FLASH_CALIB, TERM_Option_e.EXIT]
+                self.__status = option_guided[option_guided.index(self.__status) + 1]
+            else:
+                self.__status = term.queryOptions()
+
+            #Flash original program
+            if self.__status is TERM_Option_e.FLASH_ORIG:
+                log.info(f"Flashing original program")
+                self.__stm: STM_FLASH_c = STM_FLASH_c(epc_conf = _DEFAULT_EPC_CONF)
```
Comment (mariuscrsn) [inline code/MANAGER/MANAGER.py]: change to `configureDev(epc_config = _DEFAULT_EPC_CONF)`

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/MANAGER/MANAGER.py (hunk 11 above)
Comment (mariuscrsn) [reply code/MANAGER/MANAGER.py]: remove it, on flash original binary is not necessary

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Pull request merged**